core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/source/scan/text/TextScanSourceOpExec.scala

```diff
@@ -0,0 +1,53 @@
+package edu.uci.ics.texera.workflow.operators.source.scan.text
+
+import edu.uci.ics.texera.workflow.common.operators.source.SourceOperatorExecutor
+import edu.uci.ics.texera.workflow.common.tuple.Tuple
+import edu.uci.ics.texera.workflow.common.tuple.schema.Schema
+
+import java.io.{BufferedReader, FileReader, IOException}
+import java.nio.charset.StandardCharsets
+import java.nio.file.{Files, Path, Paths}
+import scala.jdk.CollectionConverters.asScalaIteratorConverter
+
+class TextScanSourceOpExec private[text] (
+    val desc: TextScanSourceOpDesc,
+    val startOffset: Int,
+    val endOffset: Int,
+    val outputAsSingleTuple: Boolean
+) extends SourceOperatorExecutor {
+  private var schema: Schema = _
+  private var reader: BufferedReader = _
+  private var rows: Iterator[String] = _
+  private var path: Path = _
+
+  @throws[IOException]
+  override def produceTexeraTuple(): Iterator[Tuple] = {
+    if (outputAsSingleTuple) {
+      Iterator(
+        Tuple
+          .newBuilder(schema)
+          .add(
+            schema.getAttribute("file"),
+            new String(Files.readAllBytes(path), StandardCharsets.UTF_8)
```

Comment:
do we have to encode it as "UTF-8"? It's fine if we decided to use this encoding, just curious, will this encoding work if I want to read a binary file, say images/video file (e.g., .jpg or .avi)?

Reply:
@Yicong-Huang Thanks for the catch, I don't think we have to use UTF-8 necessarily, I just picked that to cover the basic cases of regular text files, but I do not think it will necessarily work for all binary files.

Would you suggest an alternative encoding that could cover all these cases, or perhaps is this something we should do by inferring the encoding based on the file type or allowing the user to specify?

Reply:
I do not have a better suggestion. I do not know how other website support uploading all kinds of files with all kinds of encodings. Since this operator is for experimental usage for now, we can clearly document the limitation (i.e., what kind of files we support, what cannot support) and move on.

Reply:
ok sounds good, thank you